src/Components/Web.JS/src/PageTitle.ts

```diff
@@ -0,0 +1,33 @@
+import { getLogicalParent, LogicalElement } from './Rendering/LogicalElements';
+
+export const PageTitle = {
+  getAndRemoveExistingTitle,
+};
+
+function getAndRemoveExistingTitle(): string | null {
+  const titleElements = document.getElementsByTagName('title');
+
+  if (titleElements.length === 0) {
+    return null;
+  }
+
+  let existingTitle: string | null = null;
+
+  for (let index = titleElements.length - 1; index >= 0; index--) {
+    const currentTitleElement = titleElements[index];
+    const previousSibling = currentTitleElement.previousSibling;
+    const isBlazorTitle = previousSibling instanceof Comment && getLogicalParent(previousSibling as unknown as LogicalElement) !== null;
```

Comment:
Since there doesn't seem to be a good way to detect when a component was prerendered (AFAIK) and avoid invoking this JS method altogether, I've instead taken an approach in which `<title>` elements rendered from Blazor components are not removed after the initial render. I will concede this is a bit of a hack and only works because we know that `<title>` will be the first and only element rendered by `<PageTitle>` components. I also considered adding a special attribute to `<title>` elements rendered from Blazor components, but thought it might not be best to clutter things that way.

I'm open to other suggestions for approaches to this. If we were able to detect prerendering in `BuildRenderTree()` (or, rather, detect when we weren't prerendering), we could simply not render any `<title>` in `HeadSection` at first, then remove all `<title>` elements in `OnAfterRenderAsync()` without needing to check if they are rendered from Blazor components, then re-render with the new `<title>`. This would be a much better way of doing things IMO, but there just doesn't seem to be a good way to check if we're prerendering on the first render pass. Someone please correct me if I'm wrong about that. 😃

Reply:
I think you're right that it's not currently convenient to detect whether you're prerendering. If that's the missing piece to unlock a better solution, we could definitely add such a mechanism. However let's not block this PR on that right now.

> not render any <title> in HeadSection at first, then remove all <title> elements in OnAfterRenderAsync() without needing to check if they are rendered from Blazor components, then re-render with the new <title>

The way you have it here looks good to me. I was trying to think through some alternate schemes like "do the JS interop call during `OnInitializedAsync` (so the outbound call arrives before the first renderbatch)" but it's not clear to me that any alternate arrangement has advantages over what you're already doing here. I'm not worried about the trick you used to detect a Blazor-rendered `<title>` since it will work today, and if we have to change how it detects it in the future, we can do so.

The one thing I'm slightly unsure about is how this is going to interact with screen readers. If there's something asynchronous about how we remove and re-add `<title>` elements, that might trigger a screen reader to re-announce the title. I'm not sure. Mainly I'd suggest not being blocked about that right now, and we'll see how it ends up looking in samples, and we can adjust any details about the startup flow in the next preview if we need to!